rosbag2_compression/test/rosbag2_compression/test_sequential_compression_writer.cpp

```diff
@@ -624,6 +625,82 @@ TEST_P(SequentialCompressionWriterTest, split_event_calls_callback_with_file_com
   }
 }
 
+TEST_F(SequentialCompressionWriterTest, snapshot_writes_to_new_file_with_file_compression)
+{
+  tmp_dir_storage_options_.max_bagfile_size = 0;
+  tmp_dir_storage_options_.max_cache_size = 200;
+  tmp_dir_storage_options_.snapshot_mode = true;
+
+  initializeFakeFileStorage();
+  // Expect a single write call when the snapshot is triggered
+  EXPECT_CALL(
+    *storage_, write(
+      An<const std::vector<std::shared_ptr<const rosbag2_storage::SerializedBagMessage>> &>())
+  ).Times(1);
+
+  rosbag2_compression::CompressionOptions compression_options {
+    DefaultTestCompressor,
+    rosbag2_compression::CompressionMode::FILE,
```

Comment:
what happens if we use `::MESSAGE` here? Looking at the logic and interaction between `SequentialCompressionWriter` and `SequentialWriter`, it should probably work just fine, but it might be worth testing it too.

Reply:
With CompressionMode::MESSAGE, the compression happens before messages will get to the cyclic buffer.
The bag split will be almost the same as in the base class. See
https://github.com/ros2/rosbag2/blob/786c3c4b8ab05271630371cf515130ba02a9cde8/rosbag2_compression/src/rosbag2_compression/sequential_compression_writer.cpp#L376-L393

IMO there is no value to test this scenario. It is basically covered by another test from the `test_sequential_writer`.

Reply:
OK, all good then.